src/pages/PlanPage/hooks/usePlanPageData.ts

```diff
@@ -0,0 +1,62 @@
+import { useQuery } from '@tanstack/react-query'
+import { useParams } from 'react-router-dom'
+import { z } from 'zod'
+
+import Api from 'shared/api'
+import { NetworkErrorObject } from 'shared/api/helpers'
+
+type URLParams = {
+  owner: string
+  provider: string
+}
+
+const RequestSchema = z.object({
+  owner: z
+    .object({
+      username: z.string().nullable(),
+      isCurrentUserPartOfOrg: z.boolean(),
+      numberOfUploads: z.number().nullable(),
+    })
+    .nullable(),
+})
+
+const query = `
+  query PlanPageData($username: String!) {
+    owner(username: $username) {
+      username
+      isCurrentUserPartOfOrg
+      numberOfUploads
+    }
+  }
+`
+
+export function usePlanPageData() {
+  const { owner, provider } = useParams<URLParams>()
```

Comment:
I'm a little hesitant to use `useParams` in these hooks, and prefer to get this info via args. The reason behind that is when you pass these values as args you can require them through TS and ensure that you actually have the values. Whereas when you just call `useParams` in the hook, you can place it anywheres without realizing that you require these values to be used.